version: '3.8'
services:
# Test DB
  testdb:
    container_name: cc-testdb
    image: mysql:8.0
    platform: linux/x86_64
    environment:
      MYSQL_ROOT_PASSWORD: mysql_root
      MYSQL_DATABASE: cc_server_test
      MYSQL_USER: mysql_user
      MYSQL_PASSWORD: mysql_password
      TZ: 'Asia/Tokyo'
    volumes:
      - ./.docker/testdb/data:/var/lib/mysql  
      - ./.docker/testdb/my.cnf:/etc/mysql/conf.d/my.cnf
      - ./.docker/testdb/sql:/docker-entrypoint-initdb.d
    tty: true
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 20s
      retries: 10

# Test
  test:
    container_name: cc-test
    environment:
      MYSQL_DATABASE: cc_server_test
      MYSQL_USER: mysql_user
      MYSQL_PASSWORD: mysql_password
      MYSQL_HOST: cc-testdb
      TEST_API_URL: http://0.0.0.0:8000
    build: 
      target: development
      context: .
      dockerfile: .docker/test/Dockerfile
    volumes:
      - ./:/go/src/app
    tty: true
    depends_on:
      testdb:
        condition: service_healthy
